projects/plugins/jetpack/_inc/client/sharing/publicize.jsx

```diff
@@ -23,6 +27,14 @@ export const Publicize = withModuleSettingsFormHelpers(
 			} );
 		}
 
+		componentDidUpdate() {
+			const isActive = this.props.getOptionValue( 'publicize' );
+			// Reload the page if Publicize is enabled.
+			if ( isActive && ! window.Initial_State.socialInitialState.is_publicize_enabled ) {
+				window.location.reload();
```

Comment:
This seems like a reasonable solution for now. We can work on making API calls for the initial state (if we want to) in the coming weeks.

Reply:
Yes, Sid created a task to work on the backend parts of it - https://github.com/Automattic/jetpack-reach/issues/310